### PR TITLE
[8/N] Fix extra warnings brought by clang-tidy-17 

### PR DESCRIPTION
**Pull request by cyyever**
Fixes #ISSUE_NUMBER

### CODE REVIEW

File: torch/csrc/cuda/Stream.cpp [hunk 1]
```diff
@@ -71,6 +71,7 @@ static PyObject* THCPStream_pynew(
 
   THCPStream* self = (THCPStream*)ptr.get();
   self->stream_id = static_cast<int64_t>(stream.id());
+  // NOLINTNEXTLINE(bugprone-signed-char-misuse)
```
Comment (Skylion007) [inline torch/csrc/cuda/Stream.cpp:74]: Ugh, this one is concerning...

File: torch/csrc/cuda/nccl.h [hunk 2]
```diff
@@ -31,8 +31,8 @@ typedef void* ncclComm_t;
 /** redefine nccl unique ID in torch scope. this should be identical to native
  * nccl impp. */
 #define NCCL_UNIQUE_ID_BYTES 128
-// NOLINTNEXTLINE(cppcoreguidelines-avoid-c-arrays,modernize-avoid-c-arrays)
 typedef struct {
+  // NOLINTNEXTLINE(*array)
   char internal[NCCL_UNIQUE_ID_BYTES];
```
[the remark this answers is not in the document]
Comment (cyyever) [reply torch/csrc/cuda/nccl.h:36]: For quick fixes, but I will review them later.

File: torch/csrc/cuda/Stream.cpp (hunk 1 above)
Comment (cyyever) [reply torch/csrc/cuda/Stream.cpp:74]: For uint8 is an alias of unsigned char...

File: torch/csrc/profiler/util.cpp [hunk 3]
```diff
@@ -312,13 +312,13 @@ std::string ivalueToStr(const c10::IValue& val, bool isString) {
     // json only takes "true" and "false" so we convert the string to lower case
     if (val.isBool()) {
       for (char& c : mystr) {
-        c = std::tolower(c);
+        c = static_cast<char>(std::tolower(c));
```
Comment (Skylion007) [inline torch/csrc/profiler/util.cpp:315]:
```suggestion
        c = std::tolower(c, std::locale());
```
from `<locale>` is the newer recommended overload of this type in modern C++

File: torch/csrc/profiler/util.cpp [hunk 4]
```diff
@@ -312,13 +312,13 @@ std::string ivalueToStr(const c10::IValue& val, bool isString) {
...
       }
     }
 
     // A double quote can cause issues with the chrome tracing so force
     // all inputs to not contain more than the 2 we add in this function
-    int count = std::count(mystr.begin(), mystr.end(), '\"');
+    auto count = std::count(mystr.begin(), mystr.end(), '\"');
```
Comment (Skylion007) [inline torch/csrc/profiler/util.cpp:321]:
```suggestion
    auto count = std::count(mystr.begin(), mystr.end(), '"');
```
Use a char overload, right? No need for a string literal?

File: torch/csrc/profiler/util.cpp (hunk 3 above)
Comment (cyyever) [reply torch/csrc/profiler/util.cpp:315]: it‘s dealing with ASCII... locale is too complicated that I usually don't touch..

**Review by ezyang — APPROVED** [review]

**Pull request closed without merging**